compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1096,6 +1096,23 @@ private boolean isTupleTypeAssignable(BType source, BType target, Set<TypePair>
         return true;
     }
 
+    private boolean checkAllTupleMembersBelongNoType(List<BType> tupleTypes) {
+        boolean isNoType = false;
+        for (BType memberType : tupleTypes) {
+            switch (memberType.tag) {
+                case TypeTags.NONE:
+                    isNoType = true;
+                    break;
+                case TypeTags.TUPLE:
+                    isNoType = checkAllTupleMembersBelongNoType(((BTupleType) memberType).tupleTypes);
+                    break;
```

Comment:
If `isNoType` here is `false` but there are subsequent member types that are noType, we will still be returning after this tuple member since we break here. Is this the expected behaviour?

Reply:
Since this breaks the switch statement it will go to the next member in the foreach?

Reply:
Ugh, sorry, my bad! But then again, the name of the check implies we're checking if all is `noType`, right? Don't we need to return false in the default case or if `isNoType` is false here? Or a subsequent iteration being `noType` will hide the non-noTypes?

Reply:
ahh.. yes! :) fixed in 946ca73